packages/sign-client/src/controllers/engine.ts

```diff
@@ -1555,18 +1555,18 @@ export class Engine extends IEngine {
         ...(sessionProperties && { sessionProperties }),
         ...(sessionConfig && { sessionConfig }),
       };
-      await this.sendResult<"wc_sessionSettle">({
-        id: payload.id,
-        topic,
-        result: true,
-        throwOnFailedPublish: true,
-      });
       const target = engineEvent("session_connect");
       const listeners = this.events.listenerCount(target);
       if (listeners === 0) {
         throw new Error(`emitting ${target} without any listeners 997`);
       }
       this.events.emit(engineEvent("session_connect"), { session });
+      await this.sendResult<"wc_sessionSettle">({
```

Comment:
This is technically still blocking the method with the `await` here. Do you think this is of concern (i.e. remove the `await`)? It would it not block the UX flow?